### PR TITLE
Allow debug optimization and symbol generation for hotload

### PR DESCRIPTION
**Pull request by MuffinTastic**
Lets us see the whole call stack and step through game code in Visual Studio

Adds two DLL dependencies:

- Microsoft.DiaSymReader
- Microsoft.DiaSymReader.PortablePdb

Not quite sure how we want to make the compilation options available to end users so I didn't go that far. Defaults to debug optimization + symbols for now

devenv_Y3sZqmP3gB

### CODE REVIEW

File: Source/Hotload/Assembly/Compiler.cs [hunk 1]
```diff
@@ -137,8 +102,90 @@ public CompileResult Compile( ProjectAssemblyInfo assemblyInfo )
 			"System.Xml.ReaderWriter.dll",
 		};
 
+	private static string s_Globals = """
```
Comment (xezno) [inline Source/Hotload/Assembly/Compiler.cs:105]: Should just be `globals`

File: Source/Hotload/Assembly/Compiler.cs [hunk 2]
```diff
@@ -137,8 +102,90 @@ public CompileResult Compile( ProjectAssemblyInfo assemblyInfo )
...
+		global using static Mocha.Common.Global;
+		""";
+
+	private static Compiler s_Instance;
```
Comment (xezno) [inline Source/Hotload/Assembly/Compiler.cs:109]: Should just be `instance`

File: Source/Hotload/Assembly/Compiler.cs [hunk 3]
```diff
@@ -1,94 +1,59 @@
...
+			compiledAssemblySymbols: compiledAssemblySymbols
+		);
+	}
+}
 
-		// System references
-		var systemReferences = new[]
+public class Compiler
+{
+	private static string[] s_SystemReferences = new[]
```
Comment (xezno) [inline Source/Hotload/Assembly/Compiler.cs:56]: Should just be `systemReferences`

**Review by xezno — CHANGES_REQUESTED** [review]
Looks good, just needs a couple of changes in terms of variable naming (static vars don't need prefixing with `s_` and private fields should be lowercase)

**Comment by MuffinTastic** [comment]
> Looks good, just needs a couple of changes in terms of variable naming (static vars don't need prefixing with `s_` and private fields should be lowercase)

Sure thing. On the `s_` prefix front I was going along with the C# conventions you linked in the README, but I did forget the camel case part. Anyways, fixing

**Comment by xezno** [comment]
Ah I wasn't aware that the s_ prefix was being used in there. In that case I should probably change everything to use that.. I will take a look at making sure everything's consistent in a bit.

This looks good to me though, should make debugging stuff a lot easier - thank you!

**Pull request merged**